src/NUnitEngine/nunit-agent/Program.cs

```diff
@@ -70,9 +69,15 @@ public static void Main(string[] args)
             LocateAgencyProcess(agencyPid);
 
             log.Info("Agent process {0} starting", pid);
+
+            // TODO: CurrentFramework throws under .NET 5.0
+#if NET5_0
+            log.Info("Running under .NET 5.0"
+#else
```

Comment:
The code was from my GUI, which does run under .NET 5.0. The CurrentFramework property, used in the code that this if excludes throws when running under .NET 5.0. I left it here assuming we would soon be adding a .NET 5.0 build, but I can remove if you prefer.

Reply:
Ahhh ok. Is a .NET 5.0 agent actually required? I was expecting we could have a single .NET Core agent that targeted netcoreapp2.1, and force that to run on different runtimes. Is that not the case? I think you're a bit further ahead on this one that we are!

Reply:
Not sure if I'm ahead or just thinking of a different audience. I'm imagining that some teams need to run their tests under multiple .NET Core versions, rather than simply trusting in backward compatibility. I think this is  a bit different from .NET Framework, because you can have side-by-side installation.

Of course, if you have e.g. 3.1 and 5.0 both installed, we may need to do something more to force use of 3.1. I'm saving that for another PR, however. OTOH if we provide only one .NET Core agent, then it has to be the lowest one we believe people will use. 

All that said, in the last commit, I removed this. :smile: We can put it back if we need it.

I'm imagining another commit to add the TCP Transport and then a third one to add whichever agents you want to include in the base release. That means the second commit won't be truly exercised, but I feel OK about that since I've already implemented it once.